File: solutions/csharp/hamming/1/Hamming.cs
public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if(firstStrand.Length != secondStrand.Length)
            throw new ArgumentException("Both sequences must be of same length");
        var errCount = 0;
        for(int i = 0; i < firstStrand.Length; i++) 
        {
            if(firstStrand[i] != secondStrand[i])
                errCount++;
        }
        return errCount;
    }
}

File: solutions/cpp/hamming/1/hamming.cpp
#include "hamming.h"
#include <stdexcept>

namespace hamming {

int compute(string a, string b) {
    if(a.length() != b.length()) {
        throw domain_error("DNA sequences must be of equal length");
    }
    int c{};
    for(unsigned long long i{}; i < a.length(); i++) {
        if(a.at(i) != b.at(i)) {
            c++;
        }
    }
    return c;
}

}  // namespace hamming
